ViewLayer/DisplayDashboardView/DisplayDashboardView.cpp

```diff
@@ -137,34 +153,19 @@ void DisplayDashboardView::packSocPercentageReceived(double packSocPercentage)
 {
     ui_.stateOfChargeCapacityWidget().setValue(packSocPercentage);
 
-    if (packSocPercentage > 85)
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(20,180,20)");
-    }
-    else if ((packSocPercentage > 70) && (packSocPercentage <= 85))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(90,190,20)");
-    }
-    else if ((packSocPercentage > 55) && (packSocPercentage <= 70))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(160,200,20)");
-    }
-    else if ((packSocPercentage > 40) && (packSocPercentage <= 55))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,210,20)");
-    }
-    else if ((packSocPercentage > 25) && (packSocPercentage <= 40))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,150,20)");
-    }
-    else if ((packSocPercentage > 10) && (packSocPercentage <= 25))
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,90,20)");
-    }
-    else
-    {
-        ui_.stateOfChargeCapacityWidget().setStyleSheet("QProgressBar::chunk:horizontal{background: rgb(240,20,20)");
-    }
+    // The rgb values for the progressbar are calculated using a two simple linear equations
+    // that were derived from defining maximum and minimum values for Red and Green values
+    // and determining a slope and an intercept. Blue will stay constant as there will be no need
+    // to change it as the progress bar will transition from Green to Red. Using this method for
+    // determining the rgb values eliminates the need for switch cases or big if-else statements.
```

Comment:
Consider writing the following instead. IMO there are alot of unnecessary details in there.

The rgb values for the progress bar are calculated with the intention of displaying a a colour closer to red for lower values and green for higher values. These are calculated using linear equations with a slope and intercept.